text/0000-ember-data-singleton-record-data.md

```diff
@@ -0,0 +1,450 @@
+argument**.
+
+## Detailed design
+
+Because `RecordData` is a public userland interface we
+must rely on capabilities reporting to handle the
+deprecations for it introduced by this RFC.
+
+We will use this opportunity to improve the encapsulation
+of `RecordData` via introduction of a `RecordDataManager`.
+
+Each new `RecordData` instance will be paired with an
+instance of a `RecordDataManager`. The manager instance will
+be responsible for handling interop, deprecations and other
+book-keeping tasks we require as needed.
+
+RecordData implementations must now specify a `version`.
+The implicit version when unspecified is `"1"` (the version
+which this RFC supplants and deprecates). Implementations
+without a version or with a version equal to `"1"` will
+receive a deprecation notice the first time an instance of
+a previously unseen `RecordData` class is encountered.
+
+Note that to comply with `SemVer`, any given release of `ember-data` will
+only have support for a maximum of two versions of `RecordData`, with
+deprecated versions being dropped following LTS releases.
+
+The APIs proposed by this RFC constitute version `"2"`
+and will be used by `ember-data` when the following is
+`true`.
+
+```js
+recordData.version === "2";
+```
+
+It is possible that once the `RecordDataManager` is in place and this upgrade to `identifiers` has
+occurred that we never need to increment this `version` again. If so, a future RFC may choose to
+deprecate the `version` property altogether once "version 1" is no longer supported.
+
+### RecordDataManager
+
+The `RecordDataManager` would implement the `RecordData` interface including all
+deprecated methods and deprecated method signatures. The manager is responsible for
+
+- Managing deprecations
+- Version translation
+- Internal Book Keeping
+
+Beginning with this RFC, `StoreWrapper.recordDataFor` will return the instance of
+a `RecordDataManager` for a given `RecordData` instead of the `RecordData` itself.
+
+This encapsulates the `RecordData` feature ensuring that communication is via public API
+and provides us the ability to manage the three concerns listed above and described below.
+
+**Managing Deprecations**
+
+When a `RecordData` method or method signature is deprecated, the `RecordDataManager` would be
+responsible for detecting calls to deprecated methods or using deprecated method signatures
+and printing the appropriate deprecation messages.
+
+**Version Translation**
+
+When a `RecordData` method or method signature is deprecated the `target` may not have
+implemented the deprecated method or signature. The `RecordDataManager` would use the
+combination of calling context, presence of the method on the `target` and the declared
+support `version` of the `target` to convert calls to deprecated methods or signatures
+into their supported modern equivalent.
+
+**Internal Book Keeping**
+
+In some situations `ember-data` will need to internally track of specific kinds of requests
```

Comment:
s/to internally track of/internally keep track of

Reply:
👍